recipes/newrelic/infrastructure/cloud/aws-linux.yml

```diff
@@ -34,12 +117,112 @@ install:
   tasks:
     default:
       cmds:
+        - task: assert_pre_req
         - task: setup
 
+    assert_pre_req:
+      cmds:
+        - |
+          # Constants
+          roleName="NewRelicInfrastructure-Integrations"
+
+          policy=$(cat <<EOT
+          {
+              "Version": "2012-10-17",
+              "Statement": [
+                  {
+                      "Sid": "",
+                      "Effect": "Allow",
+                      "Principal": {
+                          "AWS": "arn:aws:iam::754728514883:root"
+                      },
+                      "Action": "sts:AssumeRole",
+                      "Condition": {
+                          "StringEquals": {
+                              "sts:ExternalId": "{{.NEW_RELIC_ACCOUNT_ID}}"
+                          }
+                      }
+                  }
+              ]
+          }
+          EOT
+          )
+
+          createRole=$(aws iam create-role --role-name $roleName --assume-role-policy-document file://<(echo $policy) 2>&1)
+
+          getRole=$(aws iam get-role --role-name $roleName 2>&1)
+
+          if [ "$(echo "$getRole" | grep "^An error occurred (NoSuchEntity)*" | wc -l)" -eq "1" ] ; then
+              echo "$createRole" >&2
+              exit 1
+          fi
+
+          aws iam attach-role-policy --role-name $roleName --policy-arn arn:aws:iam::aws:policy/ReadOnlyAccess
+
+          policy=$(cat <<EOT
+          {
+              "Version": "2012-10-17",
+              "Statement": [
+                  {
+                  "Effect": "Allow",
+                  "Action": [
+                      "budgets:ViewBudget"
+                  ],
+                  "Resource": "*"
+                  }
+              ]
+          }
+          EOT
+          )
+
+          aws iam put-role-policy --role-name $roleName --policy-name NewRelicBudget --policy-document file://<(echo $policy)
+
     setup:
       cmds:
         - |
-          exit 131
+          # Constants
+          roleName="NewRelicInfrastructure-Integrations"
+
+          callerIdentity=$(aws sts get-caller-identity 2>&1)
+          accountName=$(echo "$callerIdentity" | sed -ne 's/.*Account": "\(.*\)",/\1/p')
+
+          # Download and fix CloudFormation template
+          curl -sLo /tmp/MetricStreams_CloudFormation.yml {{.NEW_RELIC_DOWNLOAD_URL}}cloud_integrations/aws/cloudformation/MetricStreams_CloudFormation.yml
+          sed -ie "s/AWS::AccountId,/\'AWS::AccountId\',/g" /tmp/MetricStreams_CloudFormation.yml
+
+          printf "\nDeploying CloudFormation template, this may take a few minutes...\n"
+          deploy=$(aws cloudformation deploy --template-file /tmp/MetricStreams_CloudFormation.yml --stack-name NewRelic-Metric-Stream --parameter-overrides NewRelicLicenseKey={{.NEW_RELIC_LICENSE_KEY}} --capabilities CAPABILITY_NAMED_IAM 2>&1)
+
+          if [ "$(echo "$deploy" | grep "^An error occurred (AlreadyExistsException)*" | wc -l)" -eq "1" ] ; then
```

Comment:
@noahmmcgivern we'll want to check if that stack name already exist in the pre-check, this will avoid doing a lot of work on subsequent run, which would be a frequent case after a new account first install.
We wouldn't want our status data to also be inflated with the subsequent runs, when we essentially NoOp.